### PR TITLE
Issue/228 wide images dont fit lists

### LINKED ISSUE #228: Wide images don't fit inside lists
Images that are scaled down or just match the screen width don't fit inside indented elements (lists, for example) and scroll the editor to the right:

image

### PR DESCRIPTION
**Pull request by hypest**
### Fix #228 

Dynamically scales image when line width is constraint.

### Test
1. In the demo app, put the cursor on the left of the image already there (the Aztec 1600x600px image)
2. Tap the "list" button to start a list
3. Watch the image shrink in size to fit the line/width
4. Rotate the device to landscape. Image should expand to accommodate the extra width now available
5. Rotate back to portrait. Image should shrink again

### Review
@0nko want to pick this up? Thanks!

### CODE REVIEW

File: aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt [hunk 1]
```diff
@@ -603,18 +615,15 @@ class AztecText : EditText, TextWatcher {
...
 
-            /*
-             * Following Android guidelines for keylines and spacing, screen edge margins should
-             * be 16dp.  Therefore, the width of images should be the width of the screen minus
-             * 16dp on both sides (i.e. 16 * 2 = 32).
-             *
-             * https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-baseline-grids
-             */
-            val width = context.resources.displayMetrics.widthPixels - DisplayUtils.dpToPx(context, 32)
+            // maxWidth set to the biggest of screen width/height to cater for device rotation
```
Comment (theck13) [inline aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt]: Is `maxWidth` a typo in the comment?  It seems like it should be `width`.

File: app/src/main/kotlin/org/wordpress/aztec/demo/MainActivity.kt [hunk 2]
```diff
@@ -212,9 +221,11 @@ class MainActivity : AppCompatActivity(), OnRequestPermissionsResultCallback, Vi
 
         // initialize the text & HTML
         source.displayStyledAndFormattedHtml(EXAMPLE)
-        aztec.fromHtml(source.getPureHtml())
 
-        source.history = aztec.history
+        if (savedInstanceState == null) {
+            aztec.fromHtml(source.getPureHtml())
+            source.history = aztec.history
+        }
```
Comment (theck13) [inline app/src/main/kotlin/org/wordpress/aztec/demo/MainActivity.kt]: This causes a crash if the history is not saved (i.e. `onSaveInstanceState`) and restored (i.e. `onRestoreInstanceState`) on device rotation.  Launch the app, rotate the device, tap the ***HTML*** format button, and the app crashes.  See the stack trace below.

```
02-14 09:41:46.935 7403-7403/org.wordpress.aztec E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.wordpress.aztec, PID: 7403
    kotlin.UninitializedPropertyAccessException: lateinit property history has not been initialized
        at org.wordpress.aztec.source.SourceViewEditText.beforeTextChanged(SourceViewEditText.kt:121)
        at android.widget.TextView.sendBeforeTextChanged(TextView.java:8137)
        at android.widget.TextView.setText(TextView.java:4380)
        at android.widget.TextView.setText(TextView.java:4337)
        at android.widget.EditText.setText(EditText.java:89)
        at android.widget.TextView.setText(TextView.java:4312)
        at org.wordpress.aztec.source.SourceViewEditText.displayStyledAndFormattedHtml(SourceViewEditText.kt:166)
        at org.wordpress.aztec.toolbar.AztecToolbar.toggleEditorMode(AztecToolbar.kt:179)
        at org.wordpress.aztec.demo.MainActivity.onToolbarHtmlModeClicked(MainActivity.kt:547)
        at org.wordpress.aztec.toolbar.AztecToolbar.onToolbarAction(AztecToolbar.kt:170)
        at org.wordpress.aztec.toolbar.AztecToolbar.access$onToolbarAction(AztecToolbar.kt:18)
        at org.wordpress.aztec.toolbar.AztecToolbar$initView$1.onClick(AztecToolbar.kt:96)
        at android.view.View.performClick(View.java:5637)
        at android.widget.CompoundButton.performClick(CompoundButton.java:122)
        at android.view.View$PerformClick.run(View.java:22429)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

File: aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt [hunk 3]
```diff
@@ -68,25 +64,112 @@ class AztecMediaSpan(val context: Context, private var drawable: Drawable?,
...
+        var boring: BoringLayout.Metrics? = UNKNOWN_BORING
+
+        var des = -1
+
+        if (widthMode == View.MeasureSpec.EXACTLY) {
+            // Parent has told us how big to be. So be it.
+            width = widthSize
+        } else {
+            if (des < 0) {
+                boring = BoringLayout.isBoring("", textView?.paint)
```
Comment (theck13) [inline aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt]: Ha, is this _really_ the name of a layout?!

File: aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt (hunk 1 above)
Comment (hypest) [inline aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt]: Hey @theck13 , thanks for picking this up for review!

Actually no, it's not a typo. The parameter in `imageGetter?.loadImage` is actually about the maximum image wanted to be returned. I renamed the variable with b8bfe38 to better match that meaning. Thanks!

File: app/src/main/kotlin/org/wordpress/aztec/demo/MainActivity.kt (hunk 2 above)
Comment (hypest) [inline app/src/main/kotlin/org/wordpress/aztec/demo/MainActivity.kt]: Nice catch :)

Fixed with 41e4661.

File: aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt (hunk 3 above)
Comment (hypest) [inline aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt]: Hehe, it is, yeah, and apparently a boring one :)

**Comment by hypest** [comment]
I've addressed your comments @theck13 so those are ready for another pass. But, I've seen another rotation crash (change to HTML and just rotate) so I will work on that now. You may want to postpone the next review pass until that's fixed as well. I'll let you know. Thanks!

**Pull request merged**